code/game/objects/structures/inflatable.dm

```diff
@@ -0,0 +1,328 @@
+/obj/item/inflatable
+	name = "inflatable wall"
+	desc = "A folded membrane which rapidly expands into a large cubical shape on activation."
+	icon = 'icons/obj/inflatable.dmi'
+	icon_state = "folded_wall"
+	w_class = 2
+
+/obj/item/inflatable/attack_self(mob/user)
+	playsound(loc, 'sound/items/zip.ogg', 75, 1)
+	user << "<span class='notice'>You inflate [src].</span>"
+	var/obj/structure/inflatable/R = new /obj/structure/inflatable(user.loc)
+	src.transfer_fingerprints_to(R)
+	R.add_fingerprint(user)
+	qdel(src)
+
+/obj/structure/inflatable
+	name = "inflatable wall"
+	desc = "An inflated membrane. Do not puncture."
+	density = 1
+	anchored = 1
+	opacity = 0
+
+	icon = 'icons/obj/inflatable.dmi'
+	icon_state = "wall"
+
+	var/health = 50.0
+
+/obj/structure/inflatable/New(location)
+	..()
+	air_update_turf(1)
+
+/obj/structure/inflatable/Destroy()
+	air_update_turf(1)
+	return ..()
+
+/obj/structure/inflatable/CanPass(atom/movable/mover, turf/target, height=0, air_group=0)
+	return 0
+
+/obj/structure/inflatable/CanAtmosPass(turf/T)
+	return !density
+
+/obj/structure/inflatable/bullet_act(var/obj/item/projectile/Proj)
+	health -= Proj.damage
+	..()
+	if(health <= 0)
+		deflate(1)
+	return
+
+/obj/structure/inflatable/ex_act(severity)
+	switch(severity)
+		if(1.0)
+			qdel(src)
+			return
+		if(2.0)
+			deflate(1)
+			return
+		if(3.0)
+			if(prob(50))
+				deflate(1)
+				return
+
+/obj/structure/inflatable/blob_act()
+	deflate(1)
+
+/obj/structure/inflatable/attack_hand(mob/user as mob)
+	add_fingerprint(user)
+	return
+
+/obj/structure/inflatable/proc/attack_generic(mob/user as mob, damage = 0)	//used by attack_alien, attack_animal, and attack_slime
+	health -= damage
+	if(health <= 0)
+		user.visible_message("<span class='danger'>[user] tears open [src]!</span>")
+		deflate(1)
+	else	//for nicer text~
+		user.visible_message("<span class='danger'>[user] tears at [src]!</span>")
+
+/obj/structure/inflatable/attack_alien(mob/user as mob)
+	if(islarva(user))
+		return
+	attack_generic(user, 15)
+
+/obj/structure/inflatable/attack_animal(mob/user as mob)
+	if(!isanimal(user))
+		return
+	var/mob/living/simple_animal/M = user
+	if(M.melee_damage_upper <= 0)
+		return
+	attack_generic(M, M.melee_damage_upper)
+
+/obj/structure/inflatable/attack_slime(mob/user as mob)
+	attack_generic(user, rand(10, 15))
+
+/obj/structure/inflatable/attackby(obj/item/weapon/W as obj, mob/user as mob, params)
+	user.changeNext_move(CLICK_CD_MELEE)
+	if(!istype(W))
+		return
+	if(W.is_sharp())
```

Comment:
I see your point but this a material made to withstand pressure and used for emergencies. I think it'd be bullshit if you could just use a pen, which EVERYONE gets, to break it down easily